core/src/main/java/com/orientechnologies/orient/core/db/tool/ODatabaseExport.java

```diff
@@ -69,11 +73,22 @@ public ODatabaseExport(final ODatabaseDocumentInternal iDatabase, final String i
     }
     final File f = new File(fileName);
     if (f.getParentFile() != null)
+      //noinspection ResultOfMethodCallIgnored
       f.getParentFile().mkdirs();
     if (f.exists())
+      //noinspection ResultOfMethodCallIgnored
       f.delete();
 
-    final GZIPOutputStream gzipOS = new GZIPOutputStream(new FileOutputStream(fileName), compressionBuffer) {
+    this.tempFileName = fileName + ".tmp";
+    final File tempFile = new File(tempFileName);
+    if (tempFile.getParentFile() != null)
+      //noinspection ResultOfMethodCallIgnored
+      tempFile.getParentFile().mkdirs();

```

Comment:
That is 3.0 version which supports all features of Java 8 API. Could you use new file API to create and remove directories and files? This will allow us to produce meaningful exceptions if the creation of directory is failed for some reason.